src/shogun/structure/CCSOSVM.cpp

```diff
@@ -538,7 +539,7 @@ SGSparseVector<float64_t> CCCSOSVM::find_cutting_plane(float64_t* margin)
 		if (result->psi_computed)
 		{
 			new_constraint.add(result->psi_truth);
-			result->psi_pred.scale(-1.0);
+			linalg::scale(result->psi_pred, result->psi_pred, -1.0);
```

Comment:
for example here, this is bad due to the pointer de-referencing --- imagine this is an expensive operation